src/app/rosetta/lib/construction.ml

```diff
@@ -237,9 +237,10 @@ module Derive = struct
 
     let handle ~(env : Env.T(M).t) (req : Construction_derive_request.t) =
       let open M.Let_syntax in
-      let%bind pk =
+      let hex_bytes = req.public_key.hex_bytes in
+      let%bind pk_compressed =
         let pk_or_error =
-          try Ok (Rosetta_coding.Coding.to_public_key req.public_key.hex_bytes)
+          try Ok (Rosetta_coding.Coding.to_public_key_compressed hex_bytes)
```

Comment:
should this be calling the decoder in `Public_key_compressed_direct`?

Reply:
this will as written if it is passed the correct hex bytes, right? Since it branches if the length is 64:

```
let to_public_key_compressed raw =
  let len = String.length raw in
  if len = 64 then
    (* compressed encoding *)
    Public_key_compressed_direct.decode raw
  else (* uncompressed encoding *)
    to_public_key raw |> Public_key.compress
```